api/v1alpha1/httproute_types.go

```diff
@@ -230,6 +230,22 @@ type HTTPRouteAction struct {
 	// that includes the HTTPRoute will be true.
 	ForwardTo *ForwardToTarget `json:"forwardTo" protobuf:"bytes,1,opt,name=forwardTo"`
 
+	// MirrorTo sends a copy of the request to the referenced object. A
+	// response is ignored from the MirrorTo target. The resource may be
+	// "services" (omit or use the empty string for the group), or an
+	// implementation may support other resources (for example, resource
+	// "myroutetargets" in group "networking.acme.io"). Omitting or specifying
+	// the empty string for both the resource and group indicates that the
+	// resource is "services".  If the referent cannot be found, the
+	// "InvalidRoutes" status condition on any Gateway that includes the
+	// HTTPRoute will be true.
```

Comment:
> // If the referent cannot be found, the
	// "InvalidRoutes" status condition on any Gateway that includes the
	// HTTPRoute will be true.

I find this to be a little draconian.
If someone messes up a the mirror, this affects your production traffic route as well.
Now, you would usually have other checks in place to avoid this but does it make sense to be a little user-friendly here?
Can we report the status as failed for the mirror but still satisfy ForwardTo?

I might be wrong here. I'm not sure if k8s APIs allow for this or if this is acceptable or not according to the conventions.

Reply:
@hbagdi thanks for the review. I believe the broader use case should not invalidate a route containing an invalid `MirrorTo`. I think the route should reflect a degraded status condition indicating the invalid `MirrorTo`. Since status conditions are currently undefined for `HTTPRoute`, I'm going to remove for now. xref: https://github.com/kubernetes-sigs/service-apis/issues/162 to address overall `HTTPRoute` status conditions.

Reply:
That SGTM. thanks!